packages/eas-cli/src/commands/env/update.ts

```diff
@@ -0,0 +1,209 @@
+      if (!environment) {
+        environment = await promptVariableEnvironmentAsync(nonInteractive);
+      }
+      const { appVariables: existingVariables } = await EnvironmentVariablesQuery.byAppIdAsync(
+        graphqlClient,
+        projectId,
+        environment
+      );
+      if (!name) {
+        name = await selectAsync(
+          'Select variable',
+          existingVariables.map(variable => ({
+            title: variable.name,
+            value: variable.name,
+          }))
+        );
+      }
+
+      const existingVariable = existingVariables.find(variable => variable.name === name);
+      if (!existingVariable) {
+        Log.error(`Variable with name ${name} does not exist on project ${projectDisplayName}`);
+
+        return;
+      }
+
+      if (!value) {
+        value = await promptVariableValueAsync({
+          nonInteractive,
+          required: false,
+          initial: existingVariable.value,
+        });
+        if (!value || value.length === 0) {
+          value = '';
+        }
+      }
+
+      const variable = await EnvironmentVariableMutation.createForAppAsync(
+        graphqlClient,
+        {
+          name,
+          value,
+          environment,
+          visibility,
+          overwrite: true,
+        },
+        projectId
+      );
+      if (!variable) {
+        throw new Error(
+          `Could not update variable with name ${name} on project ${projectDisplayName}`
+        );
+      }
+
+      Log.withTick(
+        `Updated ${chalk.bold(name)} with value ${chalk.bold(value)} on project ${chalk.bold(
+          projectDisplayName
+        )}.`
+      );
+    } else if (scope === EnvironmentVariableScope.Shared) {
+      const sharedVariables = await EnvironmentVariablesQuery.sharedAsync(graphqlClient, projectId);
+
+      if (!name) {
+        name = await selectAsync(
+          'Select variable',
+          sharedVariables.map(variable => ({
+            title: variable.name,
+            value: variable.name,
+          }))
+        );
+      }
+
+      const existingVariable = sharedVariables.find(variable => variable.name === name);
+      if (!existingVariable) {
+        throw new Error(
+          'Variable with this name already exists on this account. Please use a different name.'
+        );
+      }
+
+      if (!value) {
+        value = await promptVariableValueAsync({
+          nonInteractive,
+          required: false,
+          initial: existingVariable.value,
+        });
+        if (!value || value.length === 0) {
+          value = '';
+        }
+      }
+
+      if (!environment && link) {
+        environment = await promptVariableEnvironmentAsync(nonInteractive);
+      }
+
+      const variable = await EnvironmentVariableMutation.createSharedVariableAsync(
+        graphqlClient,
+        {
+          name,
+          value,
+          visibility,
+          overwrite: true,
+        },
+        ownerAccount.id
+      );
```

Comment:
Sanity check: if we overwrite the shared env var, it is still linked to places it was linked to, right?

Reply:
yes, we perform update on EnvironmentVariableEntity, without destroying it like previously.